File: ackerman' s function/ackerman.py

# Python program to illustrate Ackermann function 
  
def A(m, n, s ="% s"): 
    print(s % ("A(% d, % d)" % (m, n))) 
    if m == 0: 
        return n + 1
    if n == 0: 
        return A(m - 1, 1, s) 
    n2 = A(m, n - 1, s % ("A(% d, %% s)" % (m - 1))) 
    return A(m - 1, n2, s) 
  
print(A(1, 2)) 
  

File: ackerman' s function/ackerman.java
// Java program to illustrate Ackermann function 

class GFG 
{ 

	static int ack(int m, int n) 
	{ 
		if (m == 0) 
		{ 
			return n + 1; 
		} 
		else if((m > 0) && (n == 0)) 
		{ 
			return ack(m - 1, 1); 
		} 
		else if((m > 0) && (n > 0)) 
		{ 
			return ack(m - 1, ack(m, n - 1)); 
		}else
		return n + 1; 
	} 

	// Driver code 
	public static void main(String args[]) 
	{ 
		System.out.println(ack(1, 2)); 
	} 
} 



